test/base.js

```diff
@@ -294,6 +294,49 @@ test('Should gracefully close with a connected client', (t) => {
   })
 })
 
+test('Should gracefully close when clients attempt to connect after calling close', (t) => {
+  t.plan(5)
+
+  const fastify = Fastify()
+
+  const oldClose = fastify.server.close
+  fastify.server.close = function (cb) {
+    const ws = new WebSocket('ws://localhost:' + fastify.server.address().port)
+
+    ws.on('close', () => {
+      t.pass('client 2 closed')
+    })
+
+    ws.on('open', () => {
+      oldClose.call(this, cb)
+    })
+  }
+
+  fastify.register(fastifyWebsocket)
+
+  fastify.get('/', { websocket: true }, (connection, request) => {
+    t.pass('received client connection')
+    t.teardown(() => connection.destroy())
+    // this connection stays alive until we close the server
+  })
+
+  fastify.listen(0, (err) => {
+    t.error(err)
+
+    const ws = new WebSocket('ws://localhost:' + fastify.server.address().port)
+
+    ws.on('close', () => {
+      t.pass('client 1 closed')
+    })
+
+    ws.on('open', (chunk) => {
+      fastify.close(function (err) {
```

Comment:
The plan expects 5, even though you may think there'd be 6. Only one of these websockets actually seems to hit the `fastify.get` route above. Looking at fastify's `close`, I think it tears down the routes before the `server.close`, so I'm guessing that's what's happening here. I'm still pretty new to fastify, so maybe I did something wrong 😅